core/jvm/src/main/scala/fs2/compression.scala

```diff
@@ -229,6 +254,51 @@ object compression {
       modificationTime: Option[Instant] = None,
       fileName: Option[String] = None,
       comment: Option[String] = None
+  )(implicit SyncF: Sync[F]): Pipe[F, Byte, Byte] =
+    gzip[F](
+      bufferSize = bufferSize,
+      deflateLevel = deflateLevel,
+      deflateStrategy = deflateStrategy,
+      modificationTime = modificationTime,
+      fileName = fileName,
+      comment = comment,
+      deflateFlushMode = None
+    )
+
+  /**
+    * Returns a pipe that incrementally compresses input into the GZIP format
+    * as defined by RFC 1952 at https://www.ietf.org/rfc/rfc1952.txt. Output is
+    * compatible with the GNU utils `gunzip` utility, as well as really anything
+    * else that understands GZIP. Note, however, that the GZIP format is not
+    * "stable" in the sense that all compressors will produce identical output
+    * given identical input. Part of the header seeding is arbitrary and chosen by
+    * the compression implementation. For this reason, the exact bytes produced
+    * by this pipe will differ in insignificant ways from the exact bytes produced
+    * by a tool like the GNU utils `gzip`.
+    *
+    * @param bufferSize The buffer size which will be used to page data
+    *                   into chunks. This will be the chunk size of the
+    *                   output stream. You should set it to be equal to
+    *                   the size of the largest chunk in the input stream.
+    *                   Setting this to a size which is ''smaller'' than
+    *                   the chunks in the input stream will result in
+    *                   performance degradation of roughly 50-75%. Default
+    *                   size is 32 KB.
+    * @param deflateLevel     level the compression level (0-9)
+    * @param deflateStrategy  strategy compression strategy -- see `java.util.zip.Deflater` for details
+    * @param modificationTime optional file modification time
+    * @param fileName         optional file name
+    * @param comment          optional file comment
+    * @param deflateFlushMode flush mode -- see `java.util.zip.Deflater` for details
+    */
+  def gzip[F[_]](
```

Comment:
At the risk of overdoing it, instead of this overload, we could have a `GzipParameters` (like `TlsParameters`) class with nicely typed values (e.g., `DeflateLevel` enum instead of an `Int`) and good defaults. Then we'd be able to add params if necessary without breaking source or binary compatibility. Thoughts?

Reply:
> At the risk of overdoing it, instead of this overload, we could have a `GzipParameters` (like `TlsParameters`) class with nicely typed values (e.g., `DeflateLevel` enum instead of an `Int`) and good defaults. Then we'd be able to add params if necessary without breaking source or binary compatibility. Thoughts?

No trouble, I'd be happy to do this. Parameter class - all good. Enumerations - I guess the underling java.util.zip.(Inflater|Deflater) implementations aren't changing so this should be fine. Enumerations would betters support the ability to swap out the underlying inflater/deflater implementations and I've been curious if a pure Scala implementation of these without their synchronization and JNI calls could compete.

Reply:
Hi @mpilquist, is the updated implementation preferred for future compatibility? Are the enumerations and defaults satisfactory? Any further suggestions or thoughts? The addition of these parameter classes increases the public surface of this package object quite substantially, should it be broken up and the original four public methods aliased? Thanks.